pkg/api/rest/create_test.go

```diff
@@ -16,26 +16,4 @@ limitations under the License.
 
 package rest
 
-import (
-	"testing"
-
-	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/api/errors"
-)
-
-func TestCheckGeneratedNameError(t *testing.T) {
-	expect := errors.NewNotFound("foo", "bar")
-	if err := CheckGeneratedNameError(Services, expect, &api.Pod{}); err != expect {
-		t.Errorf("NotFoundError should be ignored: %v", err)
-	}
-
-	expect = errors.NewAlreadyExists("foo", "bar")
-	if err := CheckGeneratedNameError(Services, expect, &api.Pod{}); err != expect {
-		t.Errorf("AlreadyExists should be returned when no GenerateName field: %v", err)
-	}
-
-	expect = errors.NewAlreadyExists("foo", "bar")
-	if err := CheckGeneratedNameError(Services, expect, &api.Pod{ObjectMeta: api.ObjectMeta{GenerateName: "foo"}}); err == nil || !errors.IsServerTimeout(err) {
-		t.Errorf("expected try again later error: %v", err)
-	}
-}
+// The original "TestCheckGeneratedNameError" test depends on service strategy, is moved to the registry/service package.
```

Comment:
OK, I removed the test to registry/service/strategy_test.go, and deleted the whole file.